guides/common/modules/proc_deploying-the-ca-certificate-on-a-host-by-using-ansible-rex.adoc

```diff
@@ -0,0 +1,33 @@
+[id="deploying-the-ca-certificate-on-a-host-by-using-ansible-rex"]
+= Deploying the CA certificate on a host by using Ansible REX
+
+You can use remote execution (REX) with the Ansible provider to deploy the CA certificate.
+
+include::snip_prerequisites-deploying-ca-cert-rex.adoc[]
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Monitor* > *Jobs*.
+. Click *Run Job*.
+. From the *Job category* list, select `Ansible Commands`.
+. From the *Job template* list, select `Download and execute a script`.
+. Click *Next*.
+. Select hosts on which you want to execute the job.
+. In the *url* field, enter the following URL:
++
+[options="nowrap" subs="+quotes,verbatim,attributes"]
+----
+https://_{foreman-example-com}_/unattended/public/foreman_ca_refresh
```

Comment:
If you go this route, you must mention that for isolated hosts this must be retrieved via a Smart Proxy (AKA Capsule) because they may not be able to access Foreman.

It would be better to implement a dedicated REX job template and that the user selects. Then the user doesn't need to input any fields and we can rely on REX to properly transfer the script. Even if the certificate expired (because SSH push mode doesn't need them) and when the host is isolated.

The same goes for the Script REX procedure.

Reply:
> It would be better to implement a dedicated REX job template and that the user selects. Then the user doesn't need to input any fields and we can rely on REX to properly transfer the script. Even if the certificate expired (because SSH push mode doesn't need them) and when the host is isolated.
The same goes for the Script REX procedure.

Created a tracker to properly track it: https://projects.theforeman.org/issues/37773

Reply:
@ekohl with our dropping of API access in the reverse proxy, will isolated hosts have access to this endpoint? do we need to add this endpoint to our list of allowed endpoints?

Reply:
> @ekohl with our dropping of API access in the reverse proxy, will isolated hosts have access to this endpoint? do we need to add this endpoint to our list of allowed endpoints?

We already have the [templates module](https://github.com/theforeman/smart-proxy/tree/develop/modules/templates) which proxies the unattended templates (also for kickstarts etc). I think the [current API](https://github.com/theforeman/smart-proxy/blob/develop/modules/templates/templates_unattended_api.rb) should cover it, but not 100% sure. This needs to be verified.

Reply:
@ekohl Almost. We are missing `get "/:kind/:template" do` variant to get it working

Reply:
PR: https://github.com/theforeman/smart-proxy/pull/904